R/utility.R

```diff
@@ -248,3 +248,49 @@ validateClusterConfig <- function(clusterFilePath){
 
   TRUE
 }
+
+#' Utility function for creating an output file
+#'
+#' @param filePattern a pattern indicating which file(s) to upload
+#' @param url the destination blob or virtual directory within the Azure Storage container
+#'
+#' @export
+createOutputFile <- function(filePattern, url){
+  output <- list(
+    filePattern = filePattern,
+    destination = list(
+      container = list(
+        containerUrl = url
+      )
+    ),
+    uploadOptions = list(
+      uploadCondition = "taskCompletion"
+    )
+  )
+  
+  # Parsing url to obtain container's virtual directory path
+  azureDomain <- "blob.core.windows.net"
+  parsedStorageUrl <- strsplit(url, azureDomain)[[1]]
+  
+  baseUrl <- paste0(parsedStorageUrl[1], azureDomain)
```

Comment:
minor: Personally I like to have variable placeholders for array indexing, e.g.

accountName <- parsedValue[1]
accountKey <- parsedValue[2]

Makes the code much easier to read and follow since you know what you're expecting from the array.